src/NuGet.Clients/NuGet.Tools/PMUIStarterDelegateService.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Telemetry;
+
+namespace NuGetVSExtension
+{
+    [Export(typeof(IPMUIStarter))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class PMUIStarterDelegateService : IPMUIStarter
+    {
+        public void PMUIStarter()
+        {
+            OpenSolutionPackageManager();
+        }
+
+        public void OpenSolutionPackageManager()
```

Comment:
Do we get any telemetry to identify when PMUI is opened via the infobar?
Simpler would be to add an event here.
A more intricate idea would be to see about passing some argument via `PostExecCommand` as an indicator the infobar was used.

Reply:
I just updated the PR with the Telemetry Assessments, but this is a great question. I've been thinking about it and I don't think understanding how the PM UI was opened will benefit us. This will trigger a command Telemetry that looks exactly like the one when we open PM UI using Solution Explorer.

Even if we display the InfoBar, the user can open the PM UI using other methods, maybe he saw the gold bar but opened the PM UI another way by instinct and we want to measure vulnerabilities fixes.

Reply:
Understanding how the PM UI was opened can be beneficial, so I think we should take time to add the telemetry required for this entry point and others if needed. Ideally the entry point is an enum value that gets included in a LaunchPMUI event.